### PR TITLE
=pro #17397 PR validation must build all projects if project/** was modified

### PR DESCRIPTION
**Pull request by ktoso**
This change makes all projects build if the `project/` dir is in a changeset.

This is very important to include as for PRs like "update dependencies" we really do want to test all projects (dependency update may have broken things etc). Example PR which needs this is here: https://github.com/akka/akka/pull/17433

### CODE REVIEW

File: project/ValidatePullRequest.scala [hunk 1]
```diff
@@ -115,46 +115,54 @@ object ValidatePullRequest extends AutoPlugin {
...
+      (githubEnforcedBuildAll in ValidatePR).value || {
+
+        val log = streams.value.log
+        log.debug(s"Analysing project (for inclusion in PR validation): [${name.value}]")
+        val changedDirs = (changedDirectories in ValidatePR).value
+
+        // * if `project` dir was changes, it could affect any project (dependency bump etc)
+        // * if in any scope, any of the changed modules is within this projects dependencies, we must test it:
+        val shouldBeBuilt =
+          (changedDirs contains "project") ||
```
Comment (ktoso) [inline project/ValidatePullRequest.scala]: (this is just formatting changes + this boolean expression)

**Comment by 2m** [comment]
This PR contains 5 files that are leftovers from a conflict merge.

**Comment by ktoso** [comment]
Whoops sorry, fixed

**Comment by patriknw** [comment]
LGTM

File: project/ValidatePullRequest.scala [hunk 2]
```diff
@@ -15,6 +15,38 @@ import scala.util.matching.Regex
...
+    override def task = Zero.dependsOn(test in ValidatePR)
+    def log(projectName: String, l: Logger) = 
+      l.info(s"Building [$projectName] in quick mode, as it's dependencies were affected by PR.")
+  }
+  case object BuildProjectChangedQuick extends BuildMode {
+    override def task = Zero.dependsOn(test in ValidatePR)
+    def log(projectName: String, l: Logger) =
+      l.info(s"Building [$projectName] as the root `project/` directory was affected by this PR.")
+  }
+  case object BuildForcedAll extends BuildMode {
```
Comment (2m) [inline project/ValidatePullRequest.scala]: This `BuildMode` is not used.

Comment (ktoso) [inline project/ValidatePullRequest.scala]: Good catch, thanks - the one used is `BuildCommentForcedAll` - has a bit more info. I'll remove and merge

**Comment by 2m** [comment]
LGTM

**Comment by ktoso** [comment]
Thanks! Merging

**Pull request merged**